fetch/fetch.go

```diff
@@ -227,32 +260,34 @@ func (f *Fetch) loop() {
 			f.activeRequests[req.hash] = append(f.activeRequests[req.hash], req)
 			rLen := len(f.activeRequests)
 			f.activeReqM.Unlock()
+			f.log.Info("request added to queue %v", req.hash.ShortString())
 			if req.priority > Low {
 				f.sendBatch([]requestMessage{{req.hint, req.hash}})
 				break
 			}
 			if rLen > batchMaxSize {
-				f.requestHashBatchFromPeers() // Process the batch.
```

Comment:
should we limit concurrency here? at least make sure `requestHashBatchFromPeers` will never hang after `stop` channel got a message.